### PR TITLE
Openvpn Hardening

### PR DESCRIPTION
**Pull request by stephdl**
Opened an issue for tracking the development

https://github.com/NethServer/dev/issues/5498

### CODE REVIEW

**Comment by stephdl** [comment]
Testing case

Two new settings

- `digest` (need to be the same on the two end of the tunnel)
- `tls-version-min` `Enforce a minimum TLS version`
   only on the VPN server: accept a minimum tls policy for the control channel, option not set it is tls1.0

The `digest` part must be in the vpn client configuration that you can download.

Now you can see the `digest` and the cipher with a strong tag, at least 192 bit for cipher and 256 bit for digest. lower is tagged as weak.

- Verify that you cannot use a cipher without `CBC` for the topology P2P
- verify that the change in `tls-version-min` are reflected in the vpn server configuration
- verify that you can create a tunnel by specifying digest and cipher
- verify that the vpn is workable with different setting of `tls-version-min`

**Review by gsanchietti — COMMENTED** [review]
Everything seems to work as expected.
But before merge I would like to have a cleaner commit history.

Could you squash it in a couple of commit?
One for the template part and one for the UI?

Otherwise I will need to squash everything into one single commit.

File: root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_OpenVpnTunnels.php [hunk 1]
```diff
@@ -47,3 +48,7 @@
 $L['Topology_p2p_label'] = "P2P - One server for each client, shared key (PSK) authentication";
 $L['LocalPeer_label'] = "Local P2P IP";
 $L['RemotePeer_label'] = "Remote P2P IP";
+$L['weak_label'] = 'Weak';
+$L['strong_label'] = 'Strong';
+$L['TlsVersionMin_label'] = 'Enforce a minimum TLS version';
+$L['Cipher_Not_Compatible'] = 'Only a cipher name ending by CBC is compatible with the P2P topology';
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_OpenVpnTunnels.php]: The "name ending by" sounds too complex. Can we just write?

    P2P topology requires a CBC cipher

...assuming both "P2P" and "CBC" are comprehensible from the surrounding context.

File: root/usr/share/nethesis/NethServer/Module/OpenVpnTunnels/Servers.php [hunk 2]
```diff
@@ -59,6 +59,25 @@ public function initialize()
         parent::initialize();
     }
 
+    public function readDigests()
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Module/OpenVpnTunnels/Servers.php:62]: Alternatives to duplicating code:
- Parent's class public method `$this->getParent()->methodName()`
- Parent's class public static method `parentClass::staticMethodName()` 
- PHP trait (for complex cases) https://github.com/search?q=org%3Anethserver+trait&type=Code http://php.net/manual/it/language.oop5.traits.php

Duplicating code can be acceptable though...

**Pull request closed without merging**